examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-control-delegate.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/microwave-oven-control-server/microwave-oven-control-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class MicrowaveOvenDevice;
+
+typedef Protocols::InteractionModel::Status (MicrowaveOvenDevice::*HandleSetCookingParametersCommand)(uint8_t cookMode,
+                                                                                                      uint32_t cookTime,
+                                                                                                      uint8_t powerSetting);
+typedef Protocols::InteractionModel::Status (MicrowaveOvenDevice::*HandleAddMoreTimeCommand)(uint32_t addedCookTime);
+
+namespace MicrowaveOvenControl {
+
+// This is an application level delegate to handle microwave oven control commands according to the specific
+// business logic.
+class MicrowaveOvenControlDelegate : public MicrowaveOvenControl::Delegate
+{
+
+private:
+    MicrowaveOvenDevice * mSetCookingParametersInstance;
+    HandleSetCookingParametersCommand mSetCookingParametersCommandCallback;
```

Comment:
Why is it with a callback registration, rather than using an overridable method of MicrowaveOvenControl::Delegate?

Reply:
We think using this way is more extendable when adding a new cluster. In the future we will add MicrowaveOvenMode cluster to this application, it will take less effort while merging program because all the instances are defined in MicrowaveOvenDevice, that is more easier to do interation between 2 clusters.